storage/blob.go

```diff
@@ -6,13 +6,42 @@ package storage
 import (
 	"context"
 	"io"
+	"os"
 
 	"github.com/zeebo/errs"
 )
 
 // ErrInvalidBlobRef is returned when an blob reference is invalid
 var ErrInvalidBlobRef = errs.Class("invalid blob ref")
 
+// FormatVersion represents differing storage format version values. Changes in FormatVersion
+// might affect how a Blobs or BlobReader or BlobWriter instance works, or they might only be
+// relevant to some higher layer. A FormatVersion must be specified when writing a new blob,
+// and the blob storage interface must store that value with the blob somehow, so that the same
+// FormatVersion is returned later when reading that stored blob.
+type FormatVersion int
+
+const (
+	// FormatV0 is the identifier for storage format v0, which also corresponds to an absence of
+	// format version information.
+	FormatV0 FormatVersion = 0
+	// FormatV1 is the identifier for storage format v1
+	FormatV1 FormatVersion = 1
+)
+
+const (
+	// MaxStorageFormatVersionSupported is the highest supported storage format version. If
+	// stored blobs claim a higher storage format version than this, or a caller requests
+	// writing a storage format version other than this, this software will not know how to
```

Comment:
nit: `format version other than this` should be `format version higher than this`?

Reply:
This const serves as both "the highest supported version for reading" and "the only supported version for writing", because I don't expect we'll ever need to support writing a different version from the latest (except in test situations, of course).

So "the highest supported storage format version" and "format version other than this" are both correct, but possibly confusing. Should I elaborate, or would it make more sense to split the two roles of the const?